platforms/documentation/docs/src/docs/userguide/authoring-builds/other/custom_gradle_types.adoc

```diff
@@ -338,3 +339,114 @@ A link:{javadocPath}/org/gradle/api/DomainObjectSet.html[DomainObjectSet] simply
 Compared to `NamedDomainObjectContainer`, a `DomainObjectSet` doesn't manage the objects in the collection. They need to be created and added manually.
 
 You can create an instance using the link:{javadocPath}/org/gradle/api/model/ObjectFactory.html#domainObjectSet-java.lang.Class-[ObjectFactory.domainObjectSet()] method.
+
+[[custom_dependencies_blocks]]
+== Plugin provided `dependencies` block
+
+NOTE: Custom `dependencies` blocks are based on incubating APIs.
+
+A plugin can provide a custom `dependencies` block that allows users to declare dependencies in a type-safe and context-aware way.
+
+For instance, instead of users needing to know and use the underlying `Configuration` name to add dependencies, a custom `dependencies` block lets the plugin pick a meaningful name that
+can be used consistently.
+
+=== Adding a custom `dependencies` block
+
+To add a custom `dependencies` block, you need to create a new type that will represent the set of dependency scopes available to users.
+That new type needs to be accessible from a part of your plugin (a domain object or extension).
+Finally, the dependency scopes need to be wired back to underlying `Configuration` objects that will be used during dependency resolution.
```

Comment:
💭 A numbered list could work well for these instructions.

Reply:
Instead of numbering here, I numbered the sections after this